### ISSUE TITLE
Smart crop not keeping sentences together

### ISSUE THREAD

**Issue by CaroFG** [meilisearch/meilisearch#2548]
**Describe the bug**
Smart crop is not working as described in this [PR](https://github.com/meilisearch/milli/pull/483)

> Now we are cropping around words in the same context as matching words. This means that we will keep words that are farther from the matching words but are in the same phrase, than words that are nearer but separated by a dot

**To Reproduce**
1. Use the movies.json data set
2. Search for the term: "John Connor" with `"attributesToCrop": ["overview"]` and `"cropLength": 5`
3. The second result is the movie "Terminator Genisys", its formatted overview is: 
`"overview": "…is 2029. John Connor, leader…"`

**Expected behavior**
I would expect to return the formatted overview as follows:
`...John Connor, leader of the...`

"Respecting" the sentence, "keeping words that are farther from the matching words but are in the same phrase", and not keeping  "words that are nearer but separated by a dot".

**Meilisearch version:** 
v0.28.0rc0

**Additional context**
For the query` Dumbo` v0.28 returns: "… Dumbo and his friends discover…" whereas v0.27 returns: "…new venture, Dumbo and his…".

**Comment by curquiza** [comment]
Checked with @Kerollmops, since @ManyTheFish is not here and the bug in a complex part of the code, we’ll wait for Many’s come back to fix this bug (in 10 days). So this bug fix will not be in rc1, but in rc2 or rc3.

**Comment by ManyTheFish** [comment]
Hey @CaroFG @curquiza!
It's not "really" a bug, this issue come from the fact that a sentence for Meilisearch is: "A group of words that are not separated by a hard-separator".
Today Meilisearch considers that `,` is a  hard-separator. And so, `is 2029`, `John Connor` and `leader` are considered in different sentences for Meilisearch.

### What can we do?

We can consider changing the list of hard/soft-separator, here is the lists:

hard separators: `.` `;` `,` `!` `?` `(` `)` `[` `]` `{` `}` `|`
soft separators: `all whitespaces` `-` `_` `\` `:` `/` `\\` `@` `"` `+` `~` `=` `^` `*` `#`

([code link for developers](https://github.com/meilisearch/charabia/blob/82c9f3bb655a543b23e5ecdbb6c25e63b76e19c4/src/classifier/classifier.rs#L56-L69))

**Comment by curquiza** [comment]
Discussed with @gmourier this is not something that will change for v0.29.0 since it will involve changing our soft/hard separator list -> this can lead to too much breaking for the users we cannot anticipate.
This kind of change needs deeper investigation before doing anything.

**Comment by curquiza** [comment]
@ManyTheFish is this issue still valid since we have introduced the token separator settings?

**Comment by ManyTheFish** [comment]
No, the hard-separators are not customizable with this feature, so this problem still exists, we could think of removing the comma from the hard separators to solve this 🤔
